mitodl/ocw-hugo-themes
# feat: implement lazy loading for videojs with dynamic import

#### Pre-Flight checklist


#### What are the relevant tickets? 
#648 

#### What's this PR do?
This PR implements lazy loading for videojs and its dependencies using dynamic imports. It also uses promise-polyfill for IE11 support.

#### How should this be manually tested?
We will have to test working of videojs (including on Internet Explorer 11). You can test it on any course that has lecture videos. Make sure your cache is cleared first.
As an example, you can use [this course.](https://github.mit.edu/ocw-content-rc/9.40-spring-2018) for testing locally.

1. Checkout this branch
2. `yarn start course`
3. Go to "[Lecture Videos](http://localhost:3000/video_galleries/lecture-videos/)"
4. Play any video. Make sure it plays.

**To test for Internet Explorer 11:**
1. Login to browserstack live.
2. Select Internet Explorer 11
3. Use [this netflify link](https://ocw-hugo-themes-course-v2-pr-1089--ocw-next.netlify.app/resources/part-1-introduction-and-overview-of-business-plans/) and play the video. Make sure it plays.
